api/src/main/java/openconsensus/OpenConsensus.java

```diff
@@ -0,0 +1,49 @@
+package openconsensus;
+
+import java.util.concurrent.atomic.AtomicReference;
+import openconsensus.trace.Tracer;
+
+/**
+ * Registry provides a global access to observability objects.
+ * An implementation is responsible for setting the Registry.
+ *
+ * @author Pavol Loffay
+ */
+public class OpenConsensus {
+
+  private static final OpenConsensus INSTANCE = new OpenConsensus();
+
+  private final AtomicReference<Tracer> tracer = new AtomicReference<>();
+
+  /**
+   * Reset to the initial state.
+   */
+  public static void reset() {
+    getInstance().tracer.set(null);
+  }
+
+  /**
+   *  Returns an instance of a {@link Tracer}.
+   *
+   * @return registered tracer or null
+   */
+  public static Tracer getTracer() {
+    return getInstance().tracer.get();
+  }
+
+  /**
+   * Sets a {@link Tracer} to the registry.
+   *
+   * @param newInstance tracer to be registered
+   * @exception IllegalStateException if a tracer is already registered
```

Comment:
my understanding was that re-setting `Tracer` is one of a requested features.

Reply:
We need feedback if there is already a tracer. There is a reset method on the registry. The initialization should happen only once, therefore, returning an exception.